homeassistant/components/rest/binary_sensor.py

```diff
@@ -114,16 +125,70 @@ def is_on(self):
         """Return true if the binary sensor is on."""
         return self._is_on
 
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
     def _update_from_rest_data(self):
         """Update state from the rest data."""
-        if self.rest.data is None:
+        response = self.rest.data
+        _LOGGER.debug("Data fetched from resource: %s", response)
+
+        if response is None:
             self._is_on = False
 
-        response = self.rest.data
+        if self.rest.headers is not None:
+            # If the http request failed, headers will be None
+            content_type = self.rest.headers.get("content-type")
+
+            if content_type and (
+                content_type.startswith("text/xml")
+                or content_type.startswith("application/xml")
+                or content_type.startswith("application/xhtml+xml")
+                or content_type.startswith("application/rss+xml")
+            ):
+                try:
+                    response = json_dumps(xmltodict.parse(response))
+                    _LOGGER.debug("JSON converted from XML: %s", response)
+                except ExpatError:
+                    _LOGGER.warning(
+                        "REST xml result could not be parsed and converted to JSON"
+                    )
+                    _LOGGER.debug("Erroneous XML: %s", response)
```

Comment:
Don't double log...

Reply:
Thanks for the feedback! Should these be cleaned in the RESTful sensor code

https://github.com/home-assistant/core/blob/61ff1b786bf6d6341b7c40895fa478058532223f/homeassistant/components/rest/sensor.py#L143-L150

as well?

Preferentially in a different PR, right?

Reply:
BTW... logging as a warning might be too much... I would use debug only

Reply:
Have removed double logging and downgraded to DEBUG level